packages/veritone-react-common/src/components/MediaPlayer/VideoSource.js

```diff
@@ -29,6 +33,51 @@ export default class VideoSource extends React.Component {
 
   UNSAFE_componentWillReceiveProps(nextProps) {
     const { video, src, streams, disablePreload } = nextProps;
+    this.loadVideo(video, src, streams, disablePreload);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const {
+      video,
+      src,
+      streams,
+      disablePreload,
+      currentTime,
+      actions,
+    } = this.props;
+    if (
+      !disablePreload &&
+      video &&
+      this.state.src &&
+      this.state.src !== prevState.src
+    ) {
+      video.load();
+    }
+    if (this.state.expiredSignedUri !== prevState.expiredSignedUri) {
+      this.setState({
+        streamUri: undefined,
+      });
+      this.loadVideo(video, src, streams, disablePreload);
+      setTimeout(() => {
+        if (actions) {
+          actions.seek(currentTime);
+        }
+      }, 500);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.player) {
+      this.player.unload();
+    }
```

Comment:
the shaka-player adds listeners, to avoid memory leak, remove the listeners 
if (this.player) {
     await this.player.destroy();
}